src/modules/admin/services/admin.service.ts

```diff
@@ -118,15 +113,22 @@ export class AdminService {
 
   async recoverUserPassword(
     body: RecoverPasswordRequest,
+    caller: AuthenticatedUser,
   ): Promise<RecoverPasswordResponse> {
-    const userPki = await this.userService.fetchOrgUser(
-      body.username,
-      body.orgName,
-    );
+    // caller must be an admin
+    if (caller.userRole !== UserRole.Admin)
```

Comment:
rather than checking for user role again and again in different services, we can use Out of the Box, [Roles Based Access Control](https://docs.nestjs.com/security/authorization) by NestJS, it will just need a decorator to define which roles are authorised to perform a specific service.